spec/system/searching_spec.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "managing users", type: :system do
+  let(:local_authority) { create(:local_authority, :default) }
+  let(:user) { create(:user, :assessor, local_authority: local_authority) }
+
+  let(:other_user) do
+    create(:user, :assessor, local_authority: local_authority)
+  end
+
+  let!(:planning_application1) do
+    create(:planning_application, user: user, local_authority: local_authority)
+  end
+
+  let!(:planning_application2) do
+    create(:planning_application, user: nil, local_authority: local_authority)
+  end
+
+  let!(:planning_application3) do
+    create(
+      :planning_application,
+      user: other_user,
+      local_authority: local_authority
+    )
+  end
+
+  before { sign_in(user) }
+
+  context "when users views her own planning applications" do
+    it "allows user to search planning applications by reference" do
+      visit(planning_applications_path(q: "exclude_others"))
+      click_link("All your applications")
+
+      within("#all") do
+        expect(page).to have_content(planning_application1.reference)
+        expect(page).to have_content(planning_application2.reference)
+        expect(page).not_to have_content(planning_application3.reference)
+      end
+
+      click_button("Search")
+
+      within("#all") do
+        expect(page).to have_content("Query can't be blank")
+        expect(page).to have_content(planning_application1.reference)
+        expect(page).to have_content(planning_application2.reference)
+        expect(page).not_to have_content(planning_application3.reference)
+      end
+
+      fill_in("Find an application", with: "00100")
+      click_button("Search")
+
+      within("#all") do
+        expect(page).to have_content(planning_application1.reference)
+        expect(page).not_to have_content(planning_application2.reference)
+        expect(page).not_to have_content(planning_application3.reference)
+      end
```

Comment:
with the update we mentioned this morning would now be great to add a spec to check current_url to to see we have the params and correct tab (if you weren't gonna add already!)

Then also i think is worth visiting this url directly and seeing the correct scope for the search so we know the search link is sharable